cloudify_hostpool/config/config.py

```diff
@@ -0,0 +1,126 @@
+# #######
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+import exceptions
+import re
+import socket
+import struct
+
+import yaml
+
+default_auth = None
+
+
+# will be moved to exceptions
+class ConfigError(exceptions.Exception):
+    def __init__(self, message):
+        super(ConfigError, self).__init__(message)
+
+
+def load_config(storage, file_name):
```

Comment:
Our idea was to optimize the memory - loading very large dictionary wouldn't be very good but while host pool will be handling up to 1000s hosts it shouldn't be very large.